### PR TITLE
#150 - Logging results of recommender evaluation runs

### PR DESCRIPTION
**Pull request by reckart**
- Improve the logging of results
- Improve robustness of the training and selection tasks by catching Throwable instead of Exception at relevant points

### CODE REVIEW

**Comment by reckart** [comment]
I think the current calculation of TN/FN/TP/FP and possibly of precision/recall/f-score in the `ExtendedResult` may not be good.

E.g. considering every token on which there is no annotation in the gold standard and also no annotation in the classifier results as a TN yields "too good" results. This definitely needs to be checked, best with unit tests.

Any comments/suggestions?

https://github.com/inception-project/inception/blob/ccf039e94bc92fbdb40e219c7ae025db94b1bff0/inception-imls-core/src/main/java/de/tudarmstadt/ukp/inception/recommendation/imls/core/dataobjects/ExtendedResult.java#L74-L150

**Comment by reckart** [comment]
Jenkins, can you test this please?

File: inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/log/RecommenderEvaluationResultEventAdapter.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        return aEvent.getUser();
+    }
+    
+    @Override
+    public String getDetails(RecommenderEvaluationResultEvent aEvent)
+    {
+        try {
+            Details details = new Details();
+            
+            details.score = aEvent.getScore();
```
Comment (Rentier) [inline inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/log/RecommenderEvaluationResultEventAdapter.java]: Can we remove this commented out stuff? I do not like commented out stuff.

Comment (reckart) [reply inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/log/RecommenderEvaluationResultEventAdapter.java]: Will do before merge and then put to the follow up PR ;)

**Pull request merged**